walkRpde.js

```diff
@@ -0,0 +1,47 @@
+const util = require('util');
+const fs = require('fs');
+const path = require('path');
+const { default: axios } = require('axios');
+
+const writeFile = util.promisify(fs.writeFile);
+
+function doContinueIterating({ nextUrl, prevNextUrl, endUrl }) {
+  if (endUrl && endUrl === nextUrl) {
+    console.log(`TERMINATING as endUrl reached: ${endUrl}`);
+    return false;
+  }
+  if (nextUrl === prevNextUrl) {
+    console.log(`TERMINATING as end of feed reached: ${nextUrl}`);
+    return false;
+  }
+  return true;
+}
+
+/**
+ * @param {string} startUrl
+ * @param {object} options
+ * @param {string} [options.endUrl]
+ */
+async function walkRpde(startUrl, options) {
+  let prevNextUrl;
+  let nextUrl = startUrl;
+  let page = 0;
```

Comment:
I think in order to preserve file order we should stick with the existing behaviour of at least three digits in the file name. Maybe even bump that up to four, given how large GLL is